example/react/src/App.js

```diff
@@ -1,19 +1,21 @@
-import React, { useState, useEffect } from 'react'
+import React, { useState } from 'react'
 import { GoTrueClient } from '@supabase/gotrue-js'
 import './tailwind.output.css'
 
-const auth = new GoTrueClient()
+const GOTRUE_URL_AUTOCONFIRM = 'http://localhost:9998'
+
+const auth = new GoTrueClient({
+  url: GOTRUE_URL_AUTOCONFIRM,
+})
 
 function App() {
-  let [session, setSession] = useState(null)
+  let [session, setSession] = useState(auth.session())
   let [email, setEmail] = useState(localStorage.getItem('email') ?? '')
   let [password, setPassword] = useState('')
   let [rememberMe, setRememberMe] = useState(false)
 
-  useEffect(() => {
-    setSession(auth.session())
-    auth.onAuthStateChange((_event, session) => setSession(session))
-  }, [])
+  // Keep the session up to date
+  auth.onAuthStateChange((_event, session) => setSession(session))
```

Comment:
Once the token is refreshed, this will trigger an update to the state whenever the token is refreshed, or the user logs out.